### DISCUSSION TITLE
Extracting a double value from an embedded JSON array

### DISCUSSION THREAD

**Discussion by johnduffymsc** [nlohmann/json#2570] [Q&A] [answered]
Hi

I'm struggling to get the value of a ```double``` from within a JSON array which is part of a larger JSON object. I'm using Ubuntu 20.04 LTS with nlohmann-json3-dev 3.7.3-1.

My JSON string is of the format:

```
{"type":"PRICE","asks":[{"price":"1.23039","liquidity":1000000},{"price":"1.23041","liquidity":2000000},{"price":"1.23042","liquidity":2000000},{"price":"1.23043","liquidity":5000000}]}
```

I acquire the string via

```
json j

while (std::cin > j)...
```
I'm trying to extract the "price" of the first element of the "asks" array into a ```double```...

```
double ask = j["asks"][0]["price"].get<double>();
```
This generates the following exception...
```
terminate called after throwing an instance of 'nlohmann::detail::type_error'
  what():  [json.exception.type_error.302] type must be number, but is string
Aborted (core dumped)
```
I'm a bit stumped. Any assistance would be much appreciated.

Kind regards

**Comment by gregmarr** [comment, marked as answer]
The json you are parsing contains a string there, not a number. `"price":"1.23039"`

You can only use `get<>` to get the actual type that is stored.  It doesn't do automatic string to number type conversions.

You will need to parse the string into a number yourself.  You could theoretically use this library itself to do it by calling `json::parse` on the string, and then getting the result as a `double`

    double ask = json::parse(j["asks"][0]["price"].get<std::string>()).get<double>()

but you'd probably want some more error checking and such in there.

**Reply by johnduffymsc** [reply]
That make sense now!

Thank you.

**Reply by gregmarr** [reply]
You're welcome.
